### ISSUE TITLE
dev server sometimes requires restart after editing route.html files

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/sapper#51]
*Moved from https://github.com/sveltejs/sapper-template/issues/14 cc @1wheel*

---

I'll get an error like this on the client: 

```
process-update.js:136 [HMR] Update check failed: Error: Manifest request to /client/7f71269cd30032b87e4f.hot-update.json timed out.
    at XMLHttpRequest.request.onreadystatechange (http://localhost:3000/client/main.7f71269cd30032b87e4f.js:64:22)
handleError @ process-update.js:136
cb @ process-update.js:47
Promise rejected (async)
check @ process-update.js:84
module.exports @ process-update.js:42
processMessage @ client.js:251
handleMessage @ client.js:131
handleMessage @ client.js:94
bootstrap 7f71269cd30032b87e4f:63 Uncaught (in promise) Error: Manifest request to /client/7f71269cd30032b87e4f.hot-update.json timed out.
    at XMLHttpRequest.request.onreadystatechange (bootstrap 7f71269cd30032b87e4f:63)
request.onreadystatechange @ bootstrap 7f71269cd30032b87e4f:63
Promise rejected (async)
check @ process-update.js:81
module.exports @ process-update.js:42
processMessage @ client.js:251
handleMessage @ client.js:131
handleMessage @ client.js:94
```

And then everything hangs. API routes don't start processing requests (no `console.log` printing in the terminal) until the process is killed and then requests finish.

Don't know enough about this stack of tools to debug, but might have something to do with lack of error handling here?

```
preload({ params, query }) {
  return fetch(`/api/feed`).then(r => r.json()).then(posts => {
    return {posts} 
  })
}
```

**Comment by Rich-Harris** [comment]
Yeah, I've encountered this too, though not when I was in a position to stop and try to figure out what was happening. Are there specific steps that are guaranteed to reproduce it? Looks like it's something to do with the HMR manifest not being generated in time or something.

Without error handlers in `preload` it'll just serve a 500 if something goes wrong, it shouldn't cause a hang. I tend not to bother catching the error, I just take the 500 (though Sapper could probably do a better job of dealing with them client-side)

**Comment by Rich-Harris** [comment]
Whatever this bug was, it's most likely fixed in either 0.7 or the (soon-to-be-released 0.8), since all this stuff has been overhauled. So I'll close this.
